addons/binding/org.openhab.binding.sensebox/src/main/java/org/openhab/binding/sensebox/internal/SenseBoxAPIConnection.java

```diff
@@ -0,0 +1,151 @@
+                if ("VEML6070".equals(sensor.getSensorType())) {
+                    // "unit" is not nicely comparable, so use sensor type for now
+                    parsedData.setUvIntensity(sensor);
+                } else if ("SDS 011".equals(sensor.getSensorType())) {
+                    // "unit" is not nicely comparable, neither is type, so use sensor title for now
+                    if ("PM2.5".equals(sensor.getTitle())) {
+                        parsedData.setParticulateMatter2dot5(sensor);
+                    } else if ("PM10".equals(sensor.getTitle())) {
+                        parsedData.setParticulateMatter10(sensor);
+                    } else {
+                        logger.debug("SDS 011 sensor title is {}", sensor.getTitle());
+                    }
+                } else if ("lx".equals(sensor.getUnit())) {
+                    if (sensor.getLastMeasurement() != null) {
+                        if (!(INVALID_BRIGHTNESS.equals(sensor.getLastMeasurement().getValue()))) {
+                            parsedData.setLuminance(sensor);
+                        }
+                    }
+                } else if ("hPa".equals(sensor.getUnit())) {
+                    parsedData.setPressure(sensor);
+                } else if ("%".equals(sensor.getUnit())) {
+                    parsedData.setHumidity(sensor);
+                } else if ("°C".equals(sensor.getUnit())) {
+                    parsedData.setTemperature(sensor);
+                } else {
+                    logger.debug("    Sensor: {}", sensor);
+                    logger.debug("    Sensor unit: {}", sensor.getUnit());
+                    logger.debug("    Sensor type: {}", sensor.getSensorType());
+                    logger.debug("    Sensor LM: {}", sensor.getLastMeasurement());
+                    if (sensor.getLastMeasurement() != null) {
+                        logger.debug("    Sensor LM value: {}", sensor.getLastMeasurement().getValue());
+                        logger.debug("    Sensor LM date: '{}'", sensor.getLastMeasurement().getCreatedAt());
+                    }
+                }
+            }
+
+            SenseBoxDescriptor descriptor = new SenseBoxDescriptor();
+            descriptor.setApiUrl(query);
+            if (StringUtils.isNotEmpty(parsedData.getImage())) {
+                descriptor.setImageUrl(SENSEMAP_IMAGE_URL_BASE + "/" + parsedData.getImage());
+            }
+            descriptor.setMapUrl(SENSEMAP_MAP_URL_BASE + "/explore/" + senseBoxId);
+            parsedData.setDescriptor(descriptor);
+
+            logger.trace("=================================");
+
+            result = parsedData;
+
+        } catch (IOException e) {
+            logger.debug("IO problems while fetching data: {} / {}", query, e.getMessage());
+            // TODO how would we switch the caller to OFFLINE???
+            // updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
```

Comment:
This looks WIP, maybe you should register add a listener to this connection which can report back.